docs/contributing/worker.mdx

```diff
@@ -0,0 +1,79 @@
+---
+title: "Worker"
+description: ""
+---
+
+**Functional Requirements:**
+* Create Flow Queue to add flow execution requests from multiple sources.
+* Consume flow requests and executes flow using `engine`.
+
+**Non-Functional Requirements:**
+* Securely execute untrusted code actions written by the user.
+* Queue tasks should be durable, doesn't get lost after shutting down the server.
+* The queue should be distributed to allow scaling worker horizontally.
+
+
+**Terminology:**
+* Flow Queue: contains flow execution requests from multiple sources.
+* Flow Worker: consume flow queue and obtaining the flow results
+* [BullMQ](https://github.com/taskforcesh/bullmq): distributed queue system based on `redis` as persistence layer.
+* [Isolate](https://github.com/ioi/isolate): creates a lightweight sandboxes locally.
+
+
+## Detailed Design
+
+### Flow Queue
+
+The flow queue is based on `BullMQ`, that would make it durable since it stores the tasks on `redis`
+and the library also support repeatable tasks.
+
+The repeatable tasks are required for flows with schedule trigger.
+
+### Flow Worker
+
+Consumes the flow queue and do the following in this order:
+
+**1. Create Sandbox:**
+
+First we obtain an available sandbox number from queue, we will call isolate binary file to clean and initialize this sandbox number.
```

Comment:
- What is meant by "obtain an available sandbox number from queue"?
- What is the range of available numbers?
- How do you know that a number is available?
- What is the number of boxes available?
- How is this number calculated?

Reply:
Updated on main branch, I commit it here by mistake